Formula/volta.rb

```diff
@@ -0,0 +1,27 @@
+class Volta < Formula
+  desc "Volta: The JavaScript Launcher"
+  homepage "https://volta.sh"
+  url "https://github.com/volta-cli/volta.git", :tag => "v0.7.2"
+
+  depends_on "rust" => :build
+
+  conflicts_with "hadoop", :because => "both install `yarn` binaries"
+  conflicts_with "node", :because => "both install `node` binaries"
+  conflicts_with "npm", :because => "both install `npm` binaries"
+  conflicts_with "yarn", :because => "both install `yarn` binaries"
+
+  def install
+    ENV["VOLTA_HOME"] = prefix.to_s
+    system "./dev/unix/volta-install.sh", "--release", "--skip-setup"
+    system "chmod +x #{opt_prefix}/bin/volta"
+  end
+
+  def caveats; <<~EOS
+    You need to run `volta setup` or add `~/.volta/bin` to your $PATH before using `volta`
+  EOS
+  end
+
+  test do
+    assert_match "0.7.2", shell_output("#{bin}/volta --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.